DEVX-1881: microservices-orders: Prepares for usage of latest kafka-s…

rspurgeon opened the pull request:
…treams-examples

Right now this will require running w/ a new variable to test with
a branch of kafka-streams-examples not yet merged.  For example:

`KAFKA_STREAMS_BRANCH=DEVX-1881 ./start.sh`

The docker version of the demo currently uses a manually created kafka-streams-examples image built from the latest source on kafka-streams-examples/DEVX-1881 and hosted on Docker Hub under cnfldemos (https://hub.docker.com/repository/docker/cnfldemos/kafka-streams-examples).

I used the following example workflow to validate against CCloud:

WARN: Create/use a dedicated environment/kafka cluster for this.

* build kafka-streams-examples changes locally
`(cd kafka-streams-examples;git fetch;git checkout DEVX-1881;git pull --rebase;mvn package -DskipTests)`

* delete topics from any previous runs:
 `ccloud kafka topic list -o json | jq -r '.[].name' | xargs -I '{}' ccloud kafka topic delete '{}'`

* pre-create topics:
`TOPICS=(orders orders-enriched payments customers order-validations warehouse-inventory _confluent-monitoring);printf "%s\n" "${TOPICS[@]}" | xargs -I '{}' ccloud kafka topic create '{}'`

* This command will run the microservices
`JAR=$PWD/kafka-streams-examples/target/kafka-streams-examples-5.5.0-standalone.jar BOOTSTRAP_SERVER=pkc-12345.us-central1.cloud.confluent.cloud:9092 SCHEMA_REGISTRY_URL=https://psrc-12345.us-central1.cloud.confluent.cloud RESTPORT=10000 CONFIG_FILE=/Users/username/.ccloud/config-file ./scripts/run-services.sh > ./services.log 2>&1`

* This will read Avro payment data from the payments topic
`kafka-avro-console-consumer --consumer.config /Users/username/.ccloud/config-file --bootstrap-server pkc-12345.us-central1.cloud.confluent.cloud:9092 --property schema.registry.url=https://psrc-12345.us-central1.cloud.confluent.cloud --property schema.registry.basic.auth.user.info=SRKEY:SRSECRET --property basic.auth.credentials.source=USER_INFO --topic payments --from-beginning`

rspurgeon committed:
DEVX-1881: microservices-orders: Prepares for usage of latest kafka-streams-examples

Right now this will require running w/ a new variable to test with
a branch of kafka-streams-examples not yet merged.  For example:

KAFKA_STREAMS_BRANCH=DEVX-1881 ./start.sh

However, running the docker version is not yet possible until we publish
a version of the docker image, or the streams team releases a version
with the changes in PR #346

rspurgeon committed:
DEVX-1881: microservices-orders proper clone kafka-streams-examples

ybyzek commented:
@rspurgeon 

1. We should not merge without a Docker-ready version of this demo.  From 5.5.1-post to 5.5.2 is GA, we need an interim solution (we publish the image or demo builds the image).  After 5.5.2 is GA, the official kafka-streams-examples Docker image should be fine.
2. Now that 5.5.1-post is GA, let's retarget both PRs to 5.5.1-post instead of 5.5.0-post.
3. How do I run this against CCloud?

rspurgeon commented:
> @rspurgeon
> 
>     1. We should not merge without a Docker-ready version of this demo.  From 5.5.1-post to 5.5.2 is GA, we need an interim solution (we publish the image or demo builds the image).  After 5.5.2 is GA, the official kafka-streams-examples Docker image should be fine.
> 
>     2. Now that 5.5.1-post is GA, let's retarget both PRs to 5.5.1-post instead of 5.5.0-post.
> 
>     3. How do I run this against CCloud?

@ybyzek 
1. I'm aware of the need for Docker, I mentioned this in the PR description.  I'm investigating a solution now.  I will not merge until that's complete, but Draft PRs are often ignored so I moved it to a Full PR.
2. Ok
3. This PR does not create a ccloud version of the demo, it's only to adopt these demos to changes in the dependent microservices code.  I've successfully ran the microservices code against ccloud (which was my goal for DEVX-1881) but an automated and documented version of this is not part of this PR.

rspurgeon commented:
@ybyzek please see ammended 'tips' on validating microservices code changes against ccloud cluster and SR

rspurgeon committed:
DEVX-1881: microservices-orders docker demo works...

with the ad-hoc docker image which has been pushed to cnfldemos on DockerHub.
The demo will not work with the standard kafka-streams-example image.

rspurgeon commented:
Docker demo now works w/ a manually built version of the kafka-streams-image docker image (see PR description)

ybyzek commented on microservices-orders/.gitignore:
```diff
@@ -1 +1,3 @@
-.microservices.*
+*.pids
+*.log
```
Unnecessary with the other `.gitignore`?

ybyzek reviewed (approved):
LGTM!  (Also approved https://github.com/confluentinc/kafka-streams-examples/pull/346)

rspurgeon committed:
DEVX-1881: microservices-orders updated bootstrap-servers cli arg

ybyzek reviewed (changes requested):
@rspurgeon just realized that all the files in https://github.com/confluentinc/examples/tree/5.5.1-post/microservices-orders/exercises also need to be updated to reflect the changes in https://github.com/confluentinc/kafka-streams-examples/pull/346/files

rspurgeon commented:
> @rspurgeon just realized that all the files in https://github.com/confluentinc/examples/tree/5.5.1-post/microservices-orders/exercises also need to updated to reflect the changes in https://github.com/confluentinc/kafka-streams-examples/pull/346/files

Thank you, will address

rspurgeon committed:
DEVX-1881: microservices-orders removed unecessary ignore entry

rspurgeon committed:
DEVX-1881: microservices-orders updates exercises for latest code

from kafka-streams-examples refactor in https://github.com/confluentinc/kafka-streams-examples/pull/346

rspurgeon commented:
@ybyzek updated the excercises code.  I don't believe that any docs changes are required but please double check me on that.  Thanks for catching this miss

rspurgeon committed:
DEVX-1881: microservices-orders removed tabs in OrderDetailsService.java exercise

ybyzek commented:
@rspurgeon what's the status of this PR? Just trying to reconcile the fact that this PR is still open, but https://github.com/confluentinc/kafka-streams-examples/pull/346 is resolved and the microservices demo epic jira is basically done.

rspurgeon commented:
@ybyzek I'm going to close this, the changes made it in https://github.com/confluentinc/examples/pull/702
